CoreImportExportPlugins/test/unit/src/au/gov/asd/tac/constellation/plugins/importexport/NewAttributeDialogNGTest.java

```diff
@@ -0,0 +1,184 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * NewAttributeDialog Test
+ *
+ * @author arcturus
+ */
+public class NewAttributeDialogNGTest {
+
+    private static final Logger LOGGER = Logger.getLogger(NewAttributeDialogNGTest.class.getName());
+
+    private final FxRobot robot = new FxRobot();
+
+    public NewAttributeDialogNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        if (!FxToolkit.isFXApplicationThreadRunning()) {
+            FxToolkit.registerPrimaryStage();
+        }
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        try {
+            FxToolkit.cleanupStages();
+        } catch (TimeoutException ex) {
+            LOGGER.log(Level.WARNING, "FxToolkit timedout trying to cleanup stages", ex);
+        }
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    // TODO: commenting out as there is a Headless Exception related to Swing.
+    // Will continue investigating this.
+    /**
+     * Test of setOkButtonAction method, of class NewAttributeDialog.
+     *
+     * @throws java.lang.InterruptedException
+     * @throws java.util.concurrent.ExecutionException
+     */
+//    @Test
+//    public void testSetOkButtonAction() throws InterruptedException, ExecutionException {
+//        System.out.println("setOkButtonAction");
+//
+//        final String[] value = new String[1];
+//
+//        final Future<String> future = WaitForAsyncUtils.asyncFx(() -> {
+//            final NewAttributeDialog instance = new NewAttributeDialog();
+//            instance.setOkButtonAction(event2 -> {
+//                value[0] = instance.getLabel();
+//            });
+//            instance.showDialog();
+//            return "done";
+//        });
+//
+//        final Window dialog = getDialog(robot);
+//
+//        robot.clickOn(
+//                robot.from(dialog.getScene().getRoot())
+//                        .lookup(".text-field")
+//                        .queryAs(TextField.class)
+//        ).write("test");
+//
+//        robot.clickOn(robot.from(dialog.getScene().getRoot())
+//                .lookup(".button")
+//                .lookup(hasText("OK"))
+//                .queryAs(Button.class));
+//
+//        WaitForAsyncUtils.waitFor(future);
+//
+//        assertEquals(value[0], "test");
+//    }
+    /**
+     * Test of getType method, of class NewAttributeDialog.
+     */
+    @Test
+    public void testGetType() {
+        System.out.println("getType");
+        Platform.runLater(() -> {
+            final NewAttributeDialog instance = new NewAttributeDialog();
+            final String expResult = "string";
+            final String result = instance.getType();
+            assertEquals(result, expResult);
```

Comment:
is there a way you're able to take the assert out of the fx thread? The trouble here is that because there aren't any asserts on the main test thread, the test will pass regardless of what happens on the fx thread (including the assert)

Similar situation for the functions below it

Reply:
Great, thank you for explaining that @antares1470. I have been able to use a `CountDownLatch` to make sure the assertion runs on the main testing thread.